compiler/rustc_target/src/spec/mod.rs

```diff
@@ -899,6 +899,54 @@ supported_targets! {
     ("aarch64_be-unknown-linux-gnu_ilp32", aarch64_be_unknown_linux_gnu_ilp32),
 }
 
+/// Controls use of stack canaries.
+#[derive(Clone, Copy, Debug, PartialEq, Hash, Eq)]
+pub enum StackProtector {
```

Comment:
I don't think this is the right place to put this, as it's not actually used by target definitions. I'm not sure what the right place would be though...

Reply:
I looked through the set of dependencies of `rustc_session` to if I could address this comment, but none of the other crates seem more appropriate. The actual user is `rustc_codegen_llvm`, but this crate depends on `rustc_session` and not the other way around. One alternative could perhaps be to set move the definition within `rustc_session` itself, rationale being that the option is "owned" by the option parser?

I've left the location alone for now though, rather than finding a completely new option location, as the other options defined there are at least somewhat similar. I did move it closer to the others, though, rather than being separated from them by the long `supported_targets!` list.